src/lib.rs

```diff
@@ -607,17 +608,46 @@ impl StructuredExtendedInformation {
         13 => deprecates_fpu_cs_ds,
         // 14 - reserved
         15 => pqe,
-        // 16-17 - reserved
+        16 => avx512f,
+        17 => avx512dq,
         18 => rdseed,
         19 => adx,
         20 => smap,
-        // 21-24 - reserved
-        25 => intel_processor_trace
-        // 26-31 - reserved
+        21 => avx512ifma,
+        // 22-24 - TODO
+        25 => intel_processor_trace,
+        26 => avx512pf,
+        27 => avx512er,
+        28 => avx512cd,
+        // 29 - TODO
+        30 => avx512bw,
+        31 => avx512vl
     });
 
     bit!(ecx, {
-        0 => prefetchwt1
+        0 => prefetchwt1,
+        1 => avx512vbmi,
+        // 2-4 - TODO
+        // 5 - reserved
+        6 => avx512vbmi2,
+        // 7 - reserved
+        // 8-10 - TODO
+        11 => avx512vnni,
+        12 => avx512bitalg,
+        // 13 - reserved
+        14 => avx512vpopcntdq
```

Comment:
Where are these coming from?

> Bits 16 - 5: Reserved.

Reply:
The Linux kernel: https://patchwork.kernel.org/patch/10033543/

Note how the Intel devs themselves are unclear about the naming scheme, some are separated with an underscore, some are not. They might make the spec consistent at some point.

Reply:
- `vpopcntdq`: https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=avx512vpop&expand=4104,3261,3264,3275,3278,3276,3273,3270,3267,3262,4104,2177,2176,3764,4082 

The others come from the kernel: https://patchwork.kernel.org/patch/10033543/

```c++
#define X86_FEATURE_AVX512_VBMI2 (16*32+ 6) /* Additional AVX512 Vector Bit Manipulation Instructions */
#define X86_FEATURE_GFNI	(16*32+ 8) /* Galois Field New Instructions */
#define X86_FEATURE_VAES	(16*32+ 9) /* Vector AES */
#define X86_FEATURE_VPCLMULQDQ	(16*32+ 10) /* Carry-Less Multiplication Double Quadword */
#define X86_FEATURE_AVX512_VNNI (16*32+ 11) /* Vector Neural Network Instructions */
#define X86_FEATURE_AVX512_BITALG (16*32+12) /* Support for VPOPCNT[B,W] and VPSHUF-BITQMB */
```